File: Data Structures & Algorithms/trapping-rain-water/submission-1.java
class Solution {
    public int trap(int[] height) {
        int fa=0;
        Deque<Integer> stack = new ArrayDeque<>();
        int ans=0;
        for(int i=0;i<height.length;i++){
            if(stack.isEmpty()||height[i]>=stack.peek()){
                stack.push(height[i]);
                fa+=ans;
                ans=0;
            }
            else{
                ans+=(stack.peek()-height[i]);
            }
        }
        stack = new ArrayDeque<>();
        ans = 0;
        for(int i=height.length-1;i>=0;i--){
            if(stack.isEmpty()||height[i]>stack.peek()){
                stack.push(height[i]);
                fa+=ans;
                ans=0;
            }
            else{
                ans+=(stack.peek()-height[i]);
            }
        }
        return fa;
    }
}

File: Data Structures & Algorithms/plus-one/submission-1.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int id=digits.size()-1;
        while(id>=0&&digits[id]==9){
            id--;
        }
        cout<<id<<"\n";
        if(id<0){
            vector<int> ans={1};
            for(int i=0;i<digits.size();i++) ans.push_back(0);
            return ans;
        }
        digits[id]++;
        for(int i=id+1;i<digits.size();i++) digits[i]=0;
        return digits;
    }
};
